### PR TITLE
Support logo in templates

### PR DESCRIPTION
**Pull request by jeromesimeon**
# Issue #240 

### Changes
- Adds support for logo in templates and template archives
  - Logo has to be a png file with dimensions 128x128
- Changes the way the template hash is calculated for metadata objects

### Flags
- Still needs a bit of clean up and a few tests

### CODE REVIEW

**Comment by jeromesimeon** [comment]
@dselman @algomaster99 Please review! This is the `support-logo` branch with a few fixes:
- change to how the hash is calculated for `Metadata`
- stricter check on dimensions
- some code cleanup

I'm still doing a few fixes, but wanted to give you an early pick. I could use a second pair of eyes on the `getJSON` method in the `Metadata` object.

File: packages/cicero-core/src/metadata.js [hunk 1]
```diff
@@ -324,9 +384,22 @@ class Metadata {
...
+     * Return the whole metadata content, for hashing
+     * @return {object} the content of the metadata object
+     */
+    toJSON() {
+        return {
+            'packageJson' : this.getPackageJson(),
+            'readme' : this.getREADME(),
+            'samples' : this.getSamples(),
+            'request' : this.getRequest(),
+            'logo' : JSON.stringify(this.getLogo()),
```
Comment (dselman) [inline packages/cicero-core/src/metadata.js]: I think I'd return the hash of the logo bytes here - or perhaps the logo bytes converted to a string. This is calling an implicit `toJSON` on the Buffer and I don't know how stable that is across Node versions.

Comment (jeromesimeon) [reply packages/cicero-core/src/metadata.js]: yes I'm unclear. I found some article:
https://medium.com/hackernoon/https-medium-com-amanhimself-converting-a-buffer-to-json-and-utf8-strings-in-nodejs-2150b1e3de57

Is doing the same by hand more or less reliable?

Comment (jeromesimeon) [reply packages/cicero-core/src/metadata.js]: This is documented. I'm not sure why this wouldn't work: https://nodejs.org/api/buffer.html#buffer_buf_tojson

Comment (jeromesimeon) [reply packages/cicero-core/src/metadata.js]: Note sure doing it by hand is better than relying on: https://github.com/nodejs/node/blob/cd1db2db48d28db805227e7f6ade79a106d85bc6/lib/buffer.js#L1073

**Comment by jeromesimeon** [comment]
I think this is ready to go -- except for resolution on the hash discussion.

**Comment by jeromesimeon** [comment]
> @jeromesimeon A small typographical error in `dimensions` is lurking here.

Good catch. Fixed.

File: packages/cicero-core/src/metadata.js (hunk 1 above)
Comment (jeromesimeon) [reply packages/cicero-core/src/metadata.js]: Still on this topic. Something like `buffer.toString('base64')` does seem like a good option too. @dselman is that what you had in mind?

Comment (jeromesimeon) [reply packages/cicero-core/src/metadata.js]: Finally settled on `buffer.toString('base64')` which should be stable and roundtrip. It also makes the result of `toJSON` easier to document or possibly more useful for applications.

**Pull request merged**